src/main/paths.ts

```diff
@@ -0,0 +1,209 @@
+ */
+class Win32ObsoletePaths implements Paths {
+  protected appData = process.env['APPDATA'] || path.join(os.homedir(), 'AppData', 'Roaming');
+  protected localAppData = process.env['LOCALAPPDATA'] || path.join(os.homedir(), 'AppData', 'Local');
+  get config() {
+    return path.join(this.appData, 'xdg.config', APP_NAME);
+  }
+
+  get electron() {
+    return path.join(this.appData, APP_NAME);
+  }
+
+  get logs() {
+    return path.join(this.localAppData, 'xdg.state', APP_NAME, 'logs');
+  }
+
+  get cache() {
+    return path.join(this.localAppData, 'xdg.cache', APP_NAME);
+  }
+
+  get wslDistro() {
+    return path.join(this.localAppData, 'xdg.state', APP_NAME, 'distro');
+  }
+
+  get lima(): string {
+    throw new Error('lima not available for win32');
+  }
+
+  get hyperkit(): string {
+    throw new Error('hyperkit not available for win32');
+  }
+}
+
+/**
+ * Recursively remove a directory and its contents.  Also remove any empty
+ * parent directories.  If the given path does not exist, no exception is raised.
+ * @param target The path to remove.
+ */
+function recursiveRemoveSync(target: string) {
+  try {
+    fs.rmSync(target, { recursive: true });
+  } catch (ex) {
+    if (ex.code === 'ENOENT') {
+      return;
+    }
+    throw ex;
+  }
+
+  const expectedErrors = ['ENOTEMPTY', 'EACCES', 'EBUSY', 'ENOENT', 'EPERM'];
+  const isDarwin = os.platform() === 'darwin';
+  const seen = new Set([target]);
+  let parent = path.dirname(target);
+
+  while (!seen.has(parent)) {
+    seen.add(parent);
+    if (isDarwin) {
+      const items = fs.readdirSync(parent);
+
+      if (items.length === 1 && items[0] === '.DS_Store') {
+        // on macOS, we _may_ have directories with just .DS_Store
+        const DSStorePath = path.join(parent, '.DS_Store');
+
+        try {
+          fs.rmSync(DSStorePath);
+        } catch (ex) {
+          console.error(`Error removing ${ DSStorePath }:`, ex);
+          break;
+        }
+      }
+    }
+    try {
+      fs.rmdirSync(parent);
+    } catch (ex) {
+      if (expectedErrors.includes(ex.code)) {
+        break;
+      }
+      throw ex;
+    }
+    parent = path.dirname(parent);
+  }
+}
+
+/**
+ * Try to rename an old directory name to a new one.  If the move failed,
+ * delete the old directory.
+ *
+ * @returns True if the rename occurred and succeeded.
+ */
+function tryRename(oldPath: string, newPath: string, info: string): boolean {
+  if (oldPath === newPath) {
+    return false;
+  }
+  try {
+    fs.mkdirSync(path.dirname(newPath), { recursive: true });
+    fs.renameSync(oldPath, newPath);
+    console.log(`Migrated ${ info } data from ${ oldPath } to ${ newPath }`);
+
+    return true;
+  } catch (ex) {
+    if (!['ENOENT', 'EEXIST'].includes(ex.code)) {
+      console.error(`Error moving ${ info }:`, ex);
+    } else {
+      console.log(`Could not move ${ oldPath } to ${ newPath }:`, ex);
+    }
+    recursiveRemoveSync(oldPath);
```

Comment:
Why aren't you also copying the oldPath to the new before deleting it?

Reply:
We're not doing any fancy things here (we're managing both the old and new paths), so there isn't going to be any cross-device stuff.  We really want this to be fast (it's blocking the startup path).

Reply:
Thought so, and I approved the PR. No need to change this.